### PR TITLE
AP_Camera/AP_Mount: send camera info and settings messages to GCS

### PR DESCRIPTION
**Pull request by rmackay9**
This adds basic support for AP_Camera and AP_Mount to send [CAMERA_INFORMATION](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5709) and [CAMERA_SETTINGS](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5725) messages to the ground station.  This is important for the [QGC custom camera/gimbal interface development](https://github.com/mavlink/qgroundcontrol/pull/10667).

The support is "basic" because only two of the camera drivers fill in the message details to any useful extent:

- Siyi driver provides the vendor and model names, firmware version, capability flags, recording mode and zoom percentage (see testing image below)
- MAVLinkCamV2 fully provides the CAMERA_INFORMATION message details but does not provide CAMERA_SETTINGS

Still, this is useful because the GCS work is being tested using a Siyi A8 and a follow-up PR will add support to the other drivers.

A small drive-by change is also made to the Siyi driver to display the camera firmware version (previously it only displayed the gimbal firmware version)

This has been bench tested on real hardware (Siyi A8 + CubeOrange)
testing-screenshot

### CODE REVIEW

File: libraries/AP_Camera/AP_Camera_Backend.cpp [hunk 1]
```diff
@@ -141,6 +141,46 @@ void AP_Camera_Backend::send_camera_feedback(mavlink_channel_t chan)
...
+    const uint8_t model_name[32] {};
+    const char cam_definition_uri[140] {};
+
+    // send CAMERA_INFORMATION message
+    mavlink_msg_camera_information_send(
+        chan,
+        AP_HAL::millis(),       // time_boot_ms
+        vendor_name,            // vendor_name uint8_t[32]
+        model_name,             // model_name uint8_t[32]
+        0,                      // firmware version uint32_t
```
Comment (peterbarker) [inline libraries/AP_Camera/AP_Camera_Backend.cpp:158]: See mavlink PRs for better default values for all of these.

I think we could reasonably fill a lot of these values in from json files on SD card or similar - once that jasmine parser thing is merged.

**Review by peterbarker — DISMISSED** [review]
Looks like these suggestions are move of an "if you're in the neighborhood thing, except for the default values thing.

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 2]
```diff
@@ -769,4 +773,73 @@ bool AP_Mount_Siyi::set_focus(FocusType focus_type, float focus_value)
...
 
+// send camera information message to GCS
+void AP_Mount_Siyi::send_camera_information(mavlink_channel_t chan) const
+{
+    // exit immediately if not initialised
+    if (!_initialised || !_got_firmware_version) {
+        return;
+    }
+
+    const uint8_t vendor_name[32] = "Siyi";
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp:784]: made static

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 3]
```diff
@@ -769,4 +773,73 @@ bool AP_Mount_Siyi::set_focus(FocusType focus_type, float focus_value)
...
+    uint8_t model_name[32] = "Unknown";
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp:785]: made static

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 4]
```diff
@@ -769,4 +773,73 @@ bool AP_Mount_Siyi::set_focus(FocusType focus_type, float focus_value)
...
+    uint32_t fw_version = _cam_firmware_version.major | (_cam_firmware_version.minor << 8) | (_cam_firmware_version.patch << 16);
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp:786]: made const

File: libraries/AP_Mount/AP_Mount_Siyi.cpp [hunk 5]
```diff
@@ -310,11 +310,15 @@ void AP_Mount_Siyi::process_packet()
...
 
-        // display camera firmware version
-        debug("Mount: SiyiCam fw:%u.%u.%u",
-              (unsigned)_msg_buff[_msg_buff_data_start+2],      // firmware major version
-              (unsigned)_msg_buff[_msg_buff_data_start+1],      // firmware minor version
-              (unsigned)_msg_buff[_msg_buff_data_start+0]);     // firmware revision
+        // consume and display camera firmware version
+        _cam_firmware_version = {_msg_buff[_msg_buff_data_start+2],     // firmware major version
+                                 _msg_buff[_msg_buff_data_start+1],     // firmware minor version
+                                 _msg_buff[_msg_buff_data_start+0]};    // firmware revision (aka patch)
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Siyi.cpp:316]: updated formatting.

File: libraries/AP_Camera/AP_Camera_Backend.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Camera/AP_Camera_Backend.cpp:159]: OK, I've changed the defaults to be NaN for focal_length and sensor_size_h/v

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**